src/Altinn.Apps/AppTemplates/AspNet/Altinn.App.PlatformServices/Implementation/EventsAppSI.cs

```diff
@@ -0,0 +1,93 @@
+using AltinnCore.Authentication.Utils;
+
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace Altinn.App.PlatformServices.Implementation
+{
+    /// <summary>
+    /// Represents an implementation of <see cref="IEvents"/> that acts as a client for the platform Events component.
+    /// </summary>
+    public class EventsAppSI : IEvents
+    {
+        private readonly ILogger _logger;
+        private readonly IHttpContextAccessor _httpContextAccessor;
+        private readonly AppSettings _settings;
+        private readonly HttpClient _client;
+
+        private readonly string _storageBaseAndHost;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="EventsAppSI"/> class.
+        /// </summary>
+        /// <param name="platformSettings">The platform settings.</param>
+        /// <param name="logger">A logger.</param>
+        /// <param name="httpContextAccessor">The http context accessor.</param>
+        /// <param name="httpClient">A HttpClient.</param>
+        /// <param name="settings">The application settings.</param>
+        public EventsAppSI(
+            IOptions<PlatformSettings> platformSettings,
+            ILogger<EventsAppSI> logger,
+            IHttpContextAccessor httpContextAccessor,
+            HttpClient httpClient,
+            IOptionsMonitor<AppSettings> settings)
+        {
+            _logger = logger;
+            _httpContextAccessor = httpContextAccessor;
+            _settings = settings.CurrentValue;
+            _storageBaseAndHost = platformSettings.Value.ApiStorageEndpoint;
+            httpClient.BaseAddress = new Uri(platformSettings.Value.ApiEventsEndpoint);
+            httpClient.DefaultRequestHeaders.Add(General.SubscriptionKeyHeaderName, platformSettings.Value.SubscriptionKey);
+            httpClient.DefaultRequestHeaders.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json"));
+            _client = httpClient;
+        }
+
+        /// <inheritdoc/>
+        public async Task<string> AddEvent(string eventType, Instance instance)
+        {
+            string source = $"{_storageBaseAndHost}instances/{instance.Id}";
+            CloudEvent cloudEvent = new CloudEvent
+            {
+                Subject = $"/party/{instance.InstanceOwner.PartyId}",
+                Type = eventType,
+                AlternativeSubject = instance.InstanceOwner.OrganisationNumber ?? instance.InstanceOwner.PersonNumber,
+                Time = DateTime.UtcNow,
+                SpecVersion = "1.0",
+                Source = new Uri(source)
+            };
+
+            string token =
+                JwtTokenUtil.GetTokenFromContext(_httpContextAccessor.HttpContext, _settings.RuntimeCookieName);
+
+            string serializedCloudEvent = JsonSerializer.Serialize(cloudEvent);
+
+            HttpResponseMessage response = await _client.PostAsync(
+                token,
+                "events",
```

Comment:
should be app after the change in platform

Reply:
Updating.